be/src/main/java/codesquad/todo/history/repository/JdbcHistoryRepository.java

```diff
@@ -32,14 +37,39 @@ public JdbcHistoryRepository(NamedParameterJdbcTemplate jdbcTemplate) {
 	@Override
 	public List<History> findAll() {
 		String sql =
-			"SELECT h.card_title, h.prev_column, h.next_column, h.created_at, a.id AS action_id, a.name AS action_name "
-				+ "FROM history AS h "
-				+ "JOIN action AS a ON h.action_id = a.id "
-				+ "WHERE h.is_deleted = false "
-				+ "ORDER BY h.id DESC;";
+			"SELECT id, card_title, prev_column, next_column, created_at, action_name, is_deleted, card_id "
+				+ "FROM history "
+				+ "WHERE is_deleted = false "
+				+ "ORDER BY id DESC;";
 		return Collections.unmodifiableList(jdbcTemplate.query(sql, historyRowMapper));
 	}
 
+	// todo : 예외 처리 수정
+	@Override
+	public History save(History history) {
+		String sql =
+			"INSERT INTO history(card_title, prev_column, next_column, created_at, is_deleted, action_name, card_id) "
+				+ "VALUES(:cardTitle, :prevColumn, :nextColumn, now(), :isDeleted, :actionName, :cardId)";
+		GeneratedKeyHolder keyHolder = new GeneratedKeyHolder();
+		jdbcTemplate.update(sql, new MapSqlParameterSource()
+			.addValue("cardTitle", history.getCardTitle())
+			.addValue("prevColumn", history.getPrevColumn())
+			.addValue("nextColumn", history.getNextColumn())
+			.addValue("isDeleted", history.isDeleted())
+			.addValue("actionName", history.getActionName())
+			.addValue("cardId", history.getCardId()), keyHolder);
+		long historyId = Objects.requireNonNull(keyHolder.getKey()).longValue();
+
+		return findById(historyId).orElseThrow();
+	}
+
+	@Override
+	public Optional<History> findById(Long id) {
+		String sql = "SELECT id, card_title, prev_column, next_Column, created_at, is_deleted, action_name, card_id "
+			+ "FROM history WHERE id = :id";
+		return Optional.ofNullable(jdbcTemplate.queryForObject(sql, Map.of("id", id), historyRowMapper));
```

Comment:
```suggestion
String sql = "SELECT ..."
return jdbcTemplate.query(sql, Map.of("id",id), historyRowMapper).stream().findAny();
```

stream, findAny를 사용하여 가독성있게 변경할 수 있습니다. 또한 queryForObject를 사용시 해당하는 데이가 없으면 예외가 발생하는 문제가 있습니다.